### PR TITLE
new command line interface

### PR DESCRIPTION
**Pull request by gertjanvanzwieten**
For review.

### CODE REVIEW

File: test_examples.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    else:
+      status = 'OK'
+    print( '* nprocs={}: {}'.format( __nprocs__, status ) )
+
+if failed:
+  print( 'some tests failed:', ', '.join( failed ) )
+else:
+  print( 'all tests passed' )
+
+sys.exit( len(failed) )
```
Comment (joostvanzwieten) [inline test_examples.py]: Why don't we include this in the test suite (`tests/examples.py`)?

File: examples/cutplate.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def conv( degree=1, nrefine=4 ):
 
 def unittest():
 
-  retvals = main( nelems=None, degree=2, maxrefine=2, plots=False, solvetol=0 )
+  retvals = main( nelems=None, degree=1, maxrefine=2, plots=False, solvetol=0 )
```
Comment (joostvanzwieten) [inline examples/cutplate.py]: Is this change intentional (in this commit)?

File: test_examples.py (hunk 1 above)
Comment (gertjanvanzwieten) [inline test_examples.py]: Indeed this is a placeholder, I wasn't sure if tests/ or inside setup.py would be best. The former then.

File: examples/cutplate.py (hunk 2 above)
Comment (gertjanvanzwieten) [inline examples/cutplate.py]: Yeah, it's just to suppress a warning, thought I'd sneek it in. But I'll split it out then. Purist :)

File: test_examples.py (hunk 1 above)
Comment (gertjanvanzwieten) [inline test_examples.py]: Fixed in de689ff.

File: examples/cutplate.py (hunk 2 above)
Comment (gertjanvanzwieten) [inline examples/cutplate.py]: c8893f0

File: nutils/cli.py [hunk 3]
```diff
@@ -14,82 +14,69 @@ def _githash( path, depth=0  ):
...
+      '--'+parameter.name,
+      type=str,
+      default=parameter.default,
+      metavar=parameter.name[0].upper(),
+      help=parameter.annotation if parameter.annotation is not inspect._empty else None,
+    )
+  return parser
+
+def _parse_args( functions ):
+  parser = argparse.ArgumentParser( usage='%(prog)s [{}] [--arg=X [...]]'.format( '|'.join(func.__name__ for func in functions) ) )
```
Comment (joostvanzwieten) [inline nutils/cli.py]: Minor thing: to be in line with many unix utilities (cp, mv, ls, ...) I'd suggest `'%(prog)s [{}] [OPTION]...'` as usage string.

File: nutils/cli.py [hunk 4]
```diff
@@ -42,6 +42,19 @@ def _relative_paths( basepath, path ):
...
       yield os.path.sep.join(parts[:i]), os.path.sep.join(parts[i:])
 
+def _mkbox( text, width ):
+  lines = []
+  words = text.split()
+  while words:
+    line = words.pop(0)
+    while words and len(line) + 1 + len(words[0]) <= width:
+      line += ' ' + words.pop(0)
+    lines.append( line )
```
Comment (joostvanzwieten) [inline nutils/cli.py]: Maybe this can be replaced with [textwrap]?

[textwrap]: https://docs.python.org/3.6/library/textwrap.html#textwrap.wrap

Comment (gertjanvanzwieten) [inline nutils/cli.py]: ah wow, yes.

File: nutils/cli.py (hunk 3 above)
Comment (gertjanvanzwieten) [inline nutils/cli.py]: ah sure

File: nutils/cli.py (hunk 4 above)
Comment (gertjanvanzwieten) [inline nutils/cli.py]: Fixed by pre-wrapping in 071e945.
